src/pages/member/NewMemberRegister.js

```diff
@@ -116,17 +117,17 @@ const NewMemberRegisterFormContainer = (props) => {
             gender
           });
 
-          console.log({body})
-
           fetchBackend("/users/create", "POST", body)
             .then((userResponse) => userResponse.json())
-            .then((userResponse) => {
+            .then(async (userResponse) => {
 
-              const { email, fname, lname } = userResponse.params.Item;
-              const name = `${fname} ${lname}`;
+              const { email, id } = userResponse.params.Item;
               const admin = email.substring(email.indexOf("@") + 1, email.length) === 'ubcbiztech.com';
 
-              props.setUser({ attributes: { email, name }, admin });
+              const authUser = await Auth.currentAuthenticatedUser({ bypassCache: true });
+              await Auth.updateUserAttributes(authUser, { 'custom:student_id': JSON.stringify(id) });
+              
+              props.setUser({ attributes: { ...authUser.attributes, 'custom:student_id':id }, admin });
               alert('Thanks for signing up!');
```

Comment:
Could we implement a spinner/loading state while all this is happening? Feel like with certain slow connections this could take a while and it would be nice to indicate that something is happening

Reply:
I'm trying to start a new PR on addressing small changes like this, like trying to disable the form submit buttons when the form is submitting, and doing proper re-routes and page refresh after submitting forms/deleting events, etc. So far, Formik has this cool property 'isFormSubmitting' that can be used to check but it's not perfect...